### PR TITLE
benchmarking related updates

### PR DESCRIPTION
**Pull request by eordentlich**
mostly to bring local script more inline with cloud versions and add parameterization via env vars.

### CODE REVIEW

File: python/benchmark/gen_data_distributed.py [hunk 1]
```diff
@@ -103,7 +106,7 @@ def gen_dataframe_and_meta(
 
         # Generate centers upfront.
         _, _, centers = make_blobs(
-            n_samples=0, n_features=cols, **params, return_centers=True
+            n_samples=1, n_features=cols, **params, return_centers=True
```
Comment (eordentlich) [inline python/benchmark/gen_data_distributed.py:109]: latest version of sklearn complained about this being 0

File: python/benchmark/gen_data_distributed.py [hunk 2]
```diff
@@ -375,7 +382,9 @@ def gen_dataframe_and_meta(
                     cols,
                     numPartitions=num_partitions,
                     seed=seed,
-                ).map(lambda nparray: nparray.tolist()),
+                ).map(
+                    lambda nparray: nparray.tolist()  # type: ignore
```
Comment (eordentlich) [inline python/benchmark/gen_data_distributed.py:386]: Not sure about this, but I think under spark 3.3.1, this type checking fails.

**Comment by eordentlich** [comment]
build

File: python/run_benchmark.sh [hunk 3]
```diff
@@ -10,115 +10,273 @@
...
+# if num_cols=3000 => arrow_batch_size=20000, scale linearly for smaller number of columns
+arrow_batch_size=$(( 20000 * ( ( $num_cols + 3000 - 1 ) / $num_cols ) ))
+
 
 # stop on first fail
 set -e
 
+sep="=================="
+
+# TODO adjust maxRecorsPerBatch based on num columns
```
Comment (leewyang) [inline python/run_benchmark.sh:85]: Is this comment still relevant?  (If so, there's a typo).

File: python/run_benchmark.sh [hunk 4]
```diff
@@ -10,115 +10,273 @@
...
-        --num_cpus 0 \
-        --train_path "/tmp/blobs/5k_3k_float32.parquet" \
-        --report_path "report_kmeans.csv" \
-        --spark_confs "spark.master=local[4]" \
-        --spark_confs "spark.driver.memory=128g" \
-        --spark_confs "spark.sql.execution.arrow.maxRecordsPerBatch=20000" \
+        --k 1000 \
+        --tol 1.0e-20 \
+        --maxIter 30 \
+        --initMode random \
```
Comment (leewyang) [inline python/run_benchmark.sh:157]: Any specific reason why these param values were chosen?  i.e. should we document that reason here?  Same with all other algos.

**Review by leewyang — APPROVED** [review]
Minor comments, otherwise LGTM

File: python/run_benchmark.sh (hunk 4 above)
Comment (eordentlich) [reply python/run_benchmark.sh:157]: The args now match what was used in our blog post and the cloud benchmarks.   High compute regime.

File: python/run_benchmark.sh (hunk 3 above)
Comment (eordentlich) [reply python/run_benchmark.sh]: No longer relevant.  Deleted.

**Comment by eordentlich** [comment]
build

**Review by leewyang — APPROVED** [review]

**Review by wbo4958 — APPROVED** [review]

**Pull request merged**